FactoryTest/FactoryTest.ino

```diff
@@ -0,0 +1,342 @@
+/*
+*   FactoryTest.ino
+*
+* This is the main program for the v2 PAPR board Factory Test app.
+*
+*/
+
+#include "PAPRHwDefs.h"
+#include "ButtonDebounce.h"
+#include "FanController.h"
+
+const int DELAY_100ms = 100;
+const int DELAY_200ms = 200;
+const int DELAY_500ms = 500;
+const int DELAY_1Sec = 1000;
+
+// These are all the exercises that this app performs, in the order that they are performed.
+enum Exercise {
+    begin,
+    buzzer,
+    vibrator,
+    fanMaximum,
+    fanMedium,
+    fanMinimum,
+    batteryVoltage,
+    fanRPM,
+    end
+};
+// Note - this list doesn't include exercises for the fan buttons, because the fan up button 
+// is used to cycle through the sequence of exercises so it gets plenty of exercise, and the
+// fan down button has a separate exercise that is not part of the sequence.
+
+/********************************************************************
+ * LEDs
+ ********************************************************************/
+
+// A list of all the LEDs, from left to right as they appear on the board.
+byte LEDpins[] = {
+    BATTERY_LED_1_PIN,
+    BATTERY_LED_2_PIN,
+    BATTERY_LED_3_PIN,
+    Error_LED_PIN,
+    MODE_LED_1_PIN,
+    MODE_LED_2_PIN,
+    MODE_LED_3_PIN
+};
+const int numLEDs = sizeof(LEDpins) / sizeof(byte);
+
+// Turn off all LEDs
+void allLEDsOff() {
+    for (int i = 0; i < numLEDs; i += 1) {
+        digitalWrite(LEDpins[i], HIGH);
+    }
+}
+
+// Turn a single LED on, then off again.
+void flashLED(byte pin, unsigned long duration) {
+    digitalWrite(pin, LOW);
```

Comment:
I would define LED_ON and LED_OFF to make it more readable

Reply:
Done